src/storage/compactor/src/server.rs

```diff
@@ -165,14 +165,11 @@ pub async fn compactor_serve(
     let observer_manager =
         ObserverManager::new_with_meta_client(meta_client.clone(), compactor_observer_node).await;
 
-    // use half of limit because any memory which would hold in meta-cache will be allocate by
-    // limited at first.
     let observer_join_handle = observer_manager.start().await;
-    let input_limit_bytes = compactor_memory_limit_bytes / 2;
+    // left some quota because we can not control it accurately.
+    let output_limit_bytes = compactor_memory_limit_bytes * 4 / 5;
```

Comment:
`compactor_memory_limit_bytes` is already reserved for memory during the calculation.

Reply:
Oh. Just for safe... Maybe we can remove it in next PR